src/v/reflection/test/async_adl_test.cc

```diff
@@ -43,13 +44,14 @@ SEASTAR_THREAD_TEST_CASE(test_async_adl_collection) {
     // Serialize
     iobuf out;
     auto original_vec = make_random_collection();
+    auto ovec_copy = original_vec;
     reflection::async_adl<ntp_vec>{}.to(out, std::move(original_vec)).get();
     const auto originals_hash = std::hash<iobuf>{}(out);
 
     // Deserialize
     iobuf_parser in(std::move(out));
     auto result = reflection::async_adl<ntp_vec>{}.from(in).get0();
-    BOOST_REQUIRE_EQUAL(original_vec, result);
+    BOOST_REQUIRE_EQUAL(ovec_copy, result);
```

Comment:
this fix should go into the commit that moved `original_vec` otherwise we increase the surface area of test failures during bisection.